src/risk_atlas_nexus/blocks/risk_explorer/explorer.py

```diff
@@ -451,33 +456,265 @@ def get_related_risk_incidents(
 
         if risk is not None:
             matching_risks = [risk]
-        if id is not None:
+        if risk_id is not None:
             matching_risks = list(filter(lambda risk: risk.id == risk_id, matching_risks))
 
         if len(matching_risks) > 0:
             risk: Risk = matching_risks[0]
-            risk_incidents = []
 
-            if risk.isDetectedBy is not None:
-                risk_incidents.append(risk.isDetectedBy)
+            risk_incident_instances = self._riskincidents or []
+            
+            risk_incident_instances = list(
+                filter(
+                    lambda risk_incident: risk_id in risk_incident.refersToRisk,
```

Comment:
If `risk_id` is None, it wil make the `risk_incident_instances` as empty or []. This will happen even if the `risk` at line no 457 is not None which is not what we want, right?